r/R/data.R

```diff
@@ -6,6 +6,214 @@
+            write_matrix_dir(file.path(directory, "pbmc_3k_rna_raw"))
+    } else {
+        mat <- open_matrix_dir(file.path(directory, "pbmc_3k_rna_raw"))
+    }
+    # Check if we already ran import
+    if (!file.exists(file.path(directory, "pbmc_3k_frags"))) {
+        atac_raw_url <- paste0(url_base, "pbmc_granulocyte_sorted_3k_atac_fragments.tsv.gz")
+        ensure_downloaded(file.path(directory, "pbmc_3k_10x.fragments.tsv.gz"), atac_raw_url, timeout = timeout)
+        frags <- open_fragments_10x(file.path(directory, "pbmc_3k_10x.fragments.tsv.gz")) %>%
+            write_fragments_dir(file.path(directory, "pbmc_3k_frags"))
+    } else {
+        frags <- open_fragments_dir(file.path(directory, "pbmc_3k_frags"))
+    }
+    if (filter_qc) {
+        # Download annotations for transcripts
+        transcripts <- read_gencode_transcripts(
+            file.path(directory, "references"), 
+            release = "42", 
+            transcript_choice = "MANE_Select",
+            annotation_set = "basic", 
+            features = "transcript"
+        )
+        blacklist <- read_encode_blacklist(file.path(directory, "references"), genome="hg38")
+        atac_qc <- qc_scATAC(frags, transcripts, blacklist)
+        # Filter to only cells that have at least 1000 reads on the RNA side
+        # a minimum of 1000 frag reads, and greater than 10 tss enrichment
+        pass_atac <- atac_qc %>%
+            dplyr::filter(nFrags > 1000, TSSEnrichment > 10) %>%
+            dplyr::pull(cellName)
+        pass_rna <- colnames(mat)[colSums(mat) > 1000]
+        filtered_cells <- intersect(pass_atac, pass_rna)
+        frags <- select_cells(frags, filtered_cells)
+        mat <- mat[, filtered_cells]
+    }
+    if (subset) {
+        # Subset to only genes/fragments that exist on chr4 and 11
+        genes_demo <- read_gencode_genes(
+            file.path(directory, "./references"),
+            release = "42",
+            annotation_set = "basic",
+        )
+        filtered_genes  <- genes_demo[genes_demo$chr %in% c("chr4", "chr11"),]$gene_id
+        # remove version numbers
+        filtered_genes <- gsub("\\..*", "", filtered_genes)
+        mat <- mat[which(rownames(mat) %in% filtered_genes), ]
+        frags <- frags %>% select_chromosomes(c("chr4", "chr11"))
+    }    
+    mat <- write_matrix_dir(mat, file.path(directory, mat_name), overwrite = TRUE)
+    frags <- write_fragments_dir(frags, file.path(directory, frags_name), overwrite = TRUE)
+    if (remove_input_data) {
+        unlink(file.path(directory, "pbmc_3k_10x.h5"))
+        unlink(file.path(directory, "pbmc_3k_10x.fragments.tsv.gz"))
+        unlink(file.path(directory, "pbmc_3k_rna_raw"))
+        unlink(file.path(directory, "pbmc_3k_frags"))
+    }
+    return(list(mat = mat, frags = frags))
+}
+
+#' Retrieve BPCells demo data
+#' 
+#' Functions to download matrices and fragments derived from the 10X Genomics PBMC 3k dataset,
```

Comment:
We should link to the dataset landing page since there are enough datasets that just saying PBMC 3k is a bit ambiguous